agency/detail/optional.hpp

```diff
@@ -70,7 +70,9 @@ inline void throw_bad_optional_access(const char* what_arg)
 }
 
 
+#if !(defined(__clang__) && defined(__CUDA__))
```

Comment:
Shouldn't this be something like

`#if defined(__CUDACC__) && !(defined(__clang__) && defined(__CUDA__))`

Like the llines in `tuple_impl.hpp`?